aptos-move/framework/src/natives/aggregator_natives/aggregator_v2.rs

```diff
@@ -270,6 +282,42 @@ fn native_try_sub(
     Ok(smallvec![Value::bool(success)])
 }
 
+fn native_is_at_least_impl(
+    context: &mut SafeNativeContext,
+    ty_args: Vec<Type>,
+    mut args: VecDeque<Value>,
+) -> SafeNativeResult<SmallVec<[Value; 1]>> {
+    abort_if_aggregator_api_not_enabled!(context);
+
+    debug_assert_eq!(args.len(), 2);
+    debug_assert_eq!(ty_args.len(), 1);
+    context.charge(AGGREGATOR_V2_IS_AT_LEAST_BASE)?;
+
+    let aggregator_value_ty = &ty_args[0];
+    let rhs = pop_value_by_type(aggregator_value_ty, &mut args, EUNSUPPORTED_AGGREGATOR_TYPE)?;
+    let aggregator = safely_pop_arg!(args, StructRef);
+
+    let max_value = get_aggregator_max_value(&aggregator, aggregator_value_ty)?;
+
+    let success = if let Some((resolver, mut delayed_field_data)) = get_context_data(context) {
+        let id = get_aggregator_value_as_id(&aggregator, aggregator_value_ty, resolver)?;
+        delayed_field_data.try_add_or_check_delta(
+            id,
+            max_value,
+            SignedU128::Negative(rhs),
+            resolver,
+            false,
```

Comment:
if this is the only use of 'false', and since here we only need to check if the value of the aggregator is at least rhs, wouldn't a simpler implementation that avoids modifying try_add_delta be to read lhs, and then share the code below with BoundedMath.unsigned_subtract? (the computation of lhs would be conditional - and that can be isolated and tested separately).

Reply:
we need to call resolver.delayed_field_try_add_delta_outcome to capture the restriction